### PR TITLE
Remove legacy mode for `generate_recording`

### PR DESCRIPTION
**Pull request by h-mayorquin**
The time has come.

There is nothing like the joy of deprecation and getting rid of old code.

image
Courtesy of Dall-e

### CODE REVIEW

**Comment by zm711** [comment]
Someone should really explain to Dall-e that it is really bad for the spines of books to lay flat like that.

**Review by alejoe91 — APPROVED** [review]

**Comment by DradeAW** [comment]
Here you're not deprecating but removing the function, or have I misunderstood?

**Comment by alejoe91** [comment]
it was already called `legacy`, so it was kind of born deprecated

**Comment by DradeAW** [comment]
I'm talking about the title of the PR, which sounds ambiguous to me.

I might be mistaking, but I thought

- Deprecating = We advise not to use, we will not support / drop this later
- Removing = You can't use it anymore

**Comment by h-mayorquin** [comment]
@DradeAW You are correct. It should be removing. I actually read a discussiong today in the python forums about it, mea culpa.

**Comment by samuelgarcia** [comment]
I am OK with this but I think we decided to keep 2 versions after a deprecation no ?
here it is only one.

**Comment by h-mayorquin** [comment]
@samuelgarcia no, we added a removing version at that point in time. Maybe you are confusing this with the modes of the `GeneratorRecording`? This is about the in-memory version of `generate_recording` the utility function.

File: src/spikeinterface/core/generate.py [hunk 1]
```diff
@@ -32,10 +32,9 @@ def generate_recording(
     set_probe: Optional[bool] = True,
     ndim: Optional[int] = 2,
     seed: Optional[int] = None,
-    mode: Literal["lazy", "legacy"] = "lazy",
 ) -> BaseRecording:
     """
-    Generate a recording object.
+    Generate a lazy recording object.
     Useful for testing for testing API and algos.
```
Comment (zm711) [inline src/spikeinterface/core/generate.py:38]:
```suggestion
    Useful for testing API and algos.
```
When one testing is not enough...

File: src/spikeinterface/core/generate.py [hunk 2]
```diff
@@ -52,10 +51,6 @@ def generate_recording(
         The number of dimensions of the probe, default: 2. Set to 3 to make 3 dimensional probes.
```
Comment (zm711) [inline src/spikeinterface/core/generate.py:51]:
```suggestion
        The number of dimensions of the probe, default: 2. Set to 3 to make 3 dimensional probe.
```

**Review by zm711 — COMMENTED** [review]
Since we are here two typos in the docstring that can be fixed.

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**